source/common/event/dispatcher_impl.cc

```diff
@@ -237,24 +289,14 @@ void DispatcherImpl::post(std::function<void()> callback) {
 }
 
 void DispatcherImpl::run(RunType type) {
-  run_tid_ = api_.threadFactory().currentThreadId();
+  base_.saveTid();
 
   // Flush all post callbacks before we run the event loop. We do this because there are post
   // callbacks that have to get run before the initial event loop starts running. libevent does
   // not guarantee that events are run in any particular order. So even if we post() and call
   // event_base_once() before some other event, the other event might get called first.
   runPostCallbacks();
-  base_scheduler_.run(type);
-}
-
-MonotonicTime DispatcherImpl::approximateMonotonicTime() const {
-  return approximate_monotonic_time_;
-}
-
-void DispatcherImpl::updateApproximateMonotonicTime() { updateApproximateMonotonicTimeInternal(); }
-
-void DispatcherImpl::updateApproximateMonotonicTimeInternal() {
-  approximate_monotonic_time_ = api_.timeSource().monotonicTime();
+  base_.run(type);
```

Comment:
I've moved all the functionality into the base. I decided to leave the deferred deletion list in DispatcherImpl since it doesn't need to be in the base, though the need to clear the list before running post callbacks makes for some annoying callback indirection.

Reply:
post callback execution happens frequently.  The extra expense of this indirection to call into the method that clears the deferred deletion list doesn't make sense.